api/adapters/s3.py

```diff
@@ -0,0 +1,51 @@
+import boto3
+import json
+from botocore.client import Config
+from botocore import UNSIGNED
+
+from api.adapters.base import AbstractRepositoryMetadataAdapter, AbstractRepositoryRequestHandler
+from api.adapters.utils import RepositoryType, register_adapter
+from api.models.catalog import DatasetMetadataDOC
+from api.models.user import Submission
+
+
+class _S3RequestHandler(AbstractRepositoryRequestHandler):
+    
+    def get_metadata(self, record_id: str):
+        endpoint_url = record_id.split("+")[0]
+        bucket_name = record_id.split("+")[1]
+        file_key = record_id.split("+")[2]
+
+        s3 = boto3.client('s3', config=Config(signature_version=UNSIGNED), endpoint_url=endpoint_url)
```

Comment:
Is this unsigned access would work with any s3 storage or it only works for our minIO?

Reply:
It will work for any S3 bucket path that has public access.